reactioncommerce/example-storefront
# Mobile nav design & bug fixes

Resolves #282 
Impact: **minor**
Type: **bugfix**

## Issue

The current mobile navigation implementation is incomplete and doesn't match the current designs.

## Solution

- Refactor the mobile navigation according to the designs
- Fix other related bugs, like the 3rd level of navigation items not being clickable

In duscusstion with @cassytaylor on zoom, we decided to add a header the root view of the navigation stack with a dedicated close button and brand. Before this change, the only way to close the menu was to either click on the backdrop, or navigate to page by clicking a menu item.

![screen shot 2018-10-05 at 11 17 45 am](https://user-images.githubusercontent.com/42217/46552966-1ef53680-c891-11e8-8b09-79ab6bc4c700.png)

## Breaking changes

None.

## Testing

1. Ensure that reaction is running with a sufficient number of tags, sub-tags, and sub-sub-tags. Use a data dump if necessary.
1. In the starter kit, resize to a mobile size
1. Open the mobile navigation drawer with the menu button
1. Navigate through the various menus
1. Ensure the menus behave as the designs suggest they should
